pandas/core/dtypes/concat.py

```diff
@@ -45,6 +46,8 @@ def get_dtype_kinds(l):
             # if to_concat contains different tz,
             # the result must be object dtype
             typ = str(arr.dtype)
+        elif is_range(arr):
```

Comment:
This is required so that ``Index._concat`` distinguishes between ``RangeIndex`` and ``Int64Index`` - as they should not be treated as equal (e.g. when appending). The ``range`` is indeed an arbitrary label.